File: src/main/java/no/nav/syfo/utils/Metrikk.java
package no.nav.syfo.utils;

import io.micrometer.core.instrument.MeterRegistry;
import io.micrometer.core.instrument.Tags;
import org.springframework.stereotype.Controller;

import javax.inject.Inject;

@Controller
public class Metrikk {

    private final MeterRegistry registry;

    @Inject
    public Metrikk(MeterRegistry registry) {
        this.registry = registry;
    }

    public void tellEndepunktKall(String navn) {
        registry.counter(
                navn,
                Tags.of("type", "info")
        ).increment();
    }

    public void countOutgoingReponse(String navn, Integer statusCode) {
        registry.counter(
                addPrefix(navn),
                Tags.of(
                        "type", "info",
                        "status", statusCode.toString()
                )
        ).increment();
    }

    public void tellHentSykepengesoknader403() {
        String navn = "hentSykepengesoknader.403";
        registry.counter(
                navn,
                Tags.of("type", "info")
        ).increment();
    }

    public void tellHttpKall(int kode) {
        registry.counter(
                addPrefix("httpstatus"),
                Tags.of(
                        "type", "info",
                        "kode", String.valueOf(kode)
                )
        ).increment();
    }

    private String addPrefix(String navn) {
        String METRIKK_PREFIX = "modiasyforest_";
        return METRIKK_PREFIX + navn;
    }
}

File: src/main/kotlin/no/nav/syfo/metric/Metrikk.kt
package no.nav.syfo.metric

import io.micrometer.core.instrument.MeterRegistry
import io.micrometer.core.instrument.Tags
import org.springframework.stereotype.Controller
import javax.inject.Inject

@Controller
class Metrikk @Inject constructor(
    private val registry: MeterRegistry
) {
    fun countEvent(navn: String) {
        registry.counter(
            addPrefix(navn),
            Tags.of("type", "info")
        ).increment()
    }

    fun tellEndepunktKall(navn: String) {
        registry.counter(
            navn,
            Tags.of("type", "info")
        ).increment()
    }

    fun countOutgoingReponse(navn: String, statusCode: Int) {
        registry.counter(
            addPrefix(navn),
            Tags.of(
                "type",
                "info",
                "status",
                statusCode.toString()
            )
        ).increment()
    }

    fun tellHttpKall(kode: Int) {
        registry.counter(
            addPrefix("httpstatus"),
            Tags.of(
                "type",
                "info",
                "kode",
                kode.toString()
            )
        ).increment()
    }

    private fun addPrefix(navn: String): String {
        val metricPrefix = "modiasyforest_"
        return metricPrefix + navn
    }
}
